gtars-core/src/models/region_set.rs

```diff
@@ -531,6 +532,56 @@ impl RegionSet {
         }
         total_count
     }
+
+    ///
+    /// Calculate statistics
+    ///
+    pub fn calculate_statistics(&self) -> HashMap<String, ChromosomeStats> {
+        let mut stats: HashMap<String, ChromosomeStats> = HashMap::new();
+
+        let mut regions_by_chr: HashMap<&String, Vec<&Region>> = HashMap::new();
+        for region in &self.regions {
+            regions_by_chr.entry(&region.chr).or_default().push(region);
+        }
+
+        for (chr, regions) in regions_by_chr {
+            let count = regions.len() as u32;
+            let widths: Vec<u32> = regions.iter().map(|r| r.width()).collect();
+            let minimum = *widths.iter().min().unwrap_or(&0);
+            let maximum = *widths.iter().max().unwrap_or(&0);
+
+            let earliest_position = regions.iter().map(|r| r.start).min().unwrap_or(0);
+            let end_position = regions.iter().map(|r| r.end).max().unwrap_or(0);
+
+            let mean = widths.iter().sum::<u32>() as f64 / count as f64;
+
+            let mut sorted_widths = widths.clone();
+            sorted_widths.sort_unstable();
+            let median = if count % 2 == 0 {
+                (sorted_widths[(count / 2 - 1) as usize] + sorted_widths[(count / 2) as usize])
+                    as f64
+                    / 2.0
+            } else {
+                sorted_widths[(count / 2) as usize] as f64
+            };
+
+            stats.insert(
+                chr.clone(),
+                ChromosomeStats {
+                    chromosome: chr.clone(),
```

Comment:
This field feels redundant now that I am looking at it.. the key already indicates which Chrom we are on. maybe its good to keep that, but it would save another clone if we got rid of this.

Reply:
it's ok for now